### ISSUE TITLE
Providing configuration to Docker

### ISSUE THREAD

**Issue by shockey** [swagger-api/swagger-editor#1547]
### Expected Behavior

I should be able to set Editor configuration options when using the Docker image.

### Current Behavior

The only way to provide configuration to Docker instances is via query string, i.e. `http://editor.swagger.io/?docExpansion=none`.

### Possible Solution

Expose a `CONFIG_JSON` env variable that is parsed and applied, resulting in something like `CONFIG_JSON='{"docExpansion": "none"}'`.
- More robust, but harder to deal with in practice.
- Writing JSON by hand is a pain, OTOH it's a nice solution for automated uses.

Create 1:1 env<>option bindings, resulting in something like `DOC_EXPANSION=none`.
- Prettier, but requires careful upkeep when new options are added.
- We'll also have to map UI pass-through options and keep those updated.
- Only allows string values, unless we get very clever. And clever tends to break.

### Context

Born out a discussion with @cabbonizio in #1546.

**Comment by cabbonizio** [comment]
Sounds like a solid idea.  Our DevOps team will appreciate this being able to do this via Docker config upon spinning the editor up.  Will test for you when you have something ready.  Let me know what else I can do to help.

**Comment by shockey** [comment]
@cabbonizio, awesome, thanks!

If you or someone on your team has the time to spare, a pull request that implements this would be welcome - it'll be a little while before we can get to this.

For reference, here's how Swagger-UI is doing Docker variables: https://github.com/swagger-api/swagger-ui/blob/master/docker-run.sh#L33. We have a similar `docker-run.sh` script in Swagger-Editor.

Also, cc @webron, please chime in if you lean one way or another w/r/t the possible solutions!

**Comment by jasonrhaas** [comment]
Anyone working on this currently?  It would be really nice to be able to configure the Swagger-editor defaults with docker env variables.

**Comment by MichakrawSB** [comment]
We’ve released Swagger Editor v5! 🎉 Check out the details here: [Inside the New Swagger Editor](https://swagger.io/blog/inside-the-new-swagger-editor/).
We’re closing old issues related to previous versions. If you think any of them are still relevant, please open a new issue – this helps us prioritize what matters most. Thanks for your activity! 🚀
